includes/TingSearchCqlDoctor.class.inc

```diff
@@ -0,0 +1,248 @@
+      if (!$this->string_is_cql($part)) {
+        $valid = FALSE;
+        break;
+      }
+    }
+
+    // Explode string by whitespace.
+    $expressions = explode(' ', $this->cql_string);
+    // replace keys with phrases
+    if (!empty($this->pattern)) {
+      $expressions = preg_replace($this->pattern, $this->replace, $expressions);
+    }
+
+    if ($valid) {
+      // Implode by blank.
+      return implode(' ', $expressions);
+    }
+
+    //String is not valid; implode by and.
+    return implode(' and ', $expressions);
+  }
+
+  /**
+   * Enqoute forward slashes and '-'.
+   *
+   * @return nothing
+   *   Alters private member cql_string.
+   */
+  private function escape_reserved_characters() {
+    $this->cql_string = str_replace('/', ' "/" ', $this->cql_string);
+    $this->cql_string = str_replace('-', ' "-" ', $this->cql_string);
+  }
+
+  /**
+   * Get a key for replacement in string.
+   *
+   * @return string
+   */
+  private function get_replace_key() {
+    $key_prefix = 'zxcv';
+    return $key_prefix . self::$replace_key++;
+  }
+
+  /**
+   * Handle parantheses.
+   *
+   * Look lace parantheses in string. If any found and content is not
+   * strict cql; enqoute the lot.
+   *
+   * @return nothing.
+   *   Alters private member cql_string.
+   *
+   */
+  private function fix_paranthesis() {
+    $valid_cql = TRUE;
+    //Grab content in paranthesis.
+    preg_match_all('$\(([^\(\)]*)\)$', $this->cql_string, $phrases);
+
+    if (empty($phrases[1])) {
+      // No matching paranthesis.
+      return;
+    }
+    foreach ($phrases[1] as $phrase) {
+      if (!$this->string_is_cql($phrase)) {
+        $valid_cql = FALSE;
+        break;
+      }
+    }
+    if ($valid_cql) {
+      foreach ($phrases[0] as $phrase) {
+        $this->set_replace_pattern($phrase);
+      }
+    }
+    else {
+      // one or more of the parenthesis holds invalid cql; quote the lot.
+      foreach ($phrases[0] as $phrase) {
+        $this->set_replace_pattern($phrase, TRUE);
+      }
+    }
+  }
+
+  /**
+   * Handle qoutes.
+   *
+   * Look for qouted content. Qouted content is replaced in searchstring.
+   */
+  private function fix_qoutes() {
+    // Greab qouted content.
+    preg_match_all('$["|\)]([^"]*)["\)]$', $this->cql_string, $phrases);
+    if (!empty($phrases[0])) {
+      foreach ($phrases[0] as $phrase) {
+        $this->set_replace_pattern($phrase);
+      }
+    }
+  }
+
+  /**
+   * Helper function to set a single replacement key and phrase.
+   *
+   * Adds given phrase to private member $replace. Retrieve a key for the replacement and replace given phrase in
+   * internal representation of search string with the key.
+   *
+   * @param bool $qoute_me.
+   *   If TRUE phrase is enqouted.
+   * @param string $phrase.
+   *   The phrase to add to private member $replace.
+   * @return nothing
+   *   Alters private member cql_string
+   */
+  private function set_replace_pattern($phrase, $qoute_me = FALSE) {
```

Comment:
The parameters in the doc-block for the function is swapped in relation to the function documentation.